proposed/2021/MachineReadableOutput.md

```diff
@@ -0,0 +1,797 @@
+
+```json
+{
+  "version": 1,
+  "MyProjectA": [
+    {
+      "framework": "netcoreapp3.1",
+      "topLevelPackages": [
+        {
+          "id": "Microsoft.Extensions.Primitives",
+          "requestedVersion": "[1.0.0, 5.0.0]",
+          "resolvedVersion": "1.0.0"
+        },
+        {
+          "id": "NuGet.Commands",
+          "requestedVersion": "4.8.0-preview3.5278",
+          "resolvedVersion": "4.8.0-preview3.5278"
+        },
+        {
+          "id": "Text2Xml.Lib",
+          "requestedVersion": "[1.1.2, 2.0.0)",
+          "resolvedVersion": "1.1.2"
+        }
+      ],
+      "transitivePackages": [
+        {
+          "id": "Microsoft.CSharp",
+          "resolvedVersion": "4.0.1"
+        },
+        {
+          "id": "Microsoft.NETCore.Platforms",
+          "resolvedVersion": "1.1.0"
+        },
+...
+      ]
+    }
+  ],
+  "MyProjectB": [
+    {
+      "framework": "netcoreapp3.1",
+      "topLevelPackages": [
+        {
+          "id": "NuGet.Commands",
+          "requestedVersion": "4.8.0-preview3.5278",
+          "resolvedVersion": "4.8.0-preview3.5278"
+        },
+        {
+          "id": "Text2Xml.Lib",
+          "requestedVersion": "1.1.2",
+          "resolvedVersion": "1.1.2"
+        }
+      ],
+      "transitivePackages": [
+        {
+          "id": "Microsoft.CSharp",
+          "resolvedVersion": "4.0.1"
+        },
+        {
+          "id": "Microsoft.NETCore.Platforms",
+          "resolvedVersion": "1.1.0"
+        },
+...
+      ]
+    },
+    {
+      "framework": "net5.0",
+      "topLevelPackages": [ 
+        {
+          "id": "NuGet.Commands",
+          "requestedVersion": "4.8.0-preview3.5278",
+          "resolvedVersion": "4.8.0-preview3.5278"
+        },
+        {
+          "id": "Text2Xml.Lib",
+          "requestedVersion": "1.1.2",
+          "resolvedVersion": "1.1.2"
+        }
+      ],
+      "transitivePackages": [
+        {
+          "id": "Microsoft.CSharp",
+          "resolvedVersion": "4.0.1"
+        },
+        {
+          "id": "Microsoft.NETCore.Platforms",
+          "resolvedVersion": "1.1.0"
+        },
+...
+      ]
+    }
+  ]
+}
+```
+
+### --flatten option
+
+Ability to use new `--flatten` option conjunction with `--json` option for all `dotnet list package` commands to ensure JSON-formatted output is emitted to the console for multiproject solution(including multitarget projects)and all projects and target frameworks are flattened. Same package with multiple version entry is possible. It's useful more for SBOM than single `--json` option.
+
+```dotnetcli
+dotnet list [<PROJECT>|<SOLUTION>] package [--config <SOURCE>]
+    [--deprecated]
+    [--framework <FRAMEWORK>] [--highest-minor] [--highest-patch]
+    [--include-prerelease] [--include-transitive] [--interactive]
+    [--outdated] [--source <SOURCE>] [-v|--verbosity <LEVEL>]
+    [--vulnerable]
+    [--json]
+    [--flatten]
+
+dotnet list package -h|--help
+```
+
+### dotnet list package
```

Comment:
Consider using heading levels to signal document structure. You already have a 3rd level heading `dotnet list package` example earlier in the document (line 50) under the same 2nd level heading. Therefore, from a document structure point of view, this is either redundant or using a bad header title.  I get that it's being done to have the json and text versions close to each other for easier comparison, but since almost the entire document is level 3 headings under the same level 2 heading, it's a bit harder to skim over the document and understand what the different "sections" of the design are.

Also, consider that GitHub makes all headers HTML anchors, so they can be deep linked. All the headers with the same title will have the same anchor, and therefore it will be impossible to deep link any except the first. It's a very minor issue, but I think it shows attention to detail if it's avoided.